packages/react-native-reanimated/plugin/src/workletFactory.ts

```diff
@@ -325,17 +325,44 @@ function hash(str: string): number {
   return (hash1 >>> 0) * 4096 + (hash2 >>> 0);
 }
 
-function makeWorkletName(fun: NodePath<WorkletizableFunction>): string {
+let functionId = 1;
+
+function makeWorkletName(
+  fun: NodePath<WorkletizableFunction>,
+  state: ReanimatedPluginPass
+): string {
+  let source = 'unknown_file';
+
+  if (state.file.opts.filename) {
+    const filepath = state.file.opts.filename;
+    // Take the file name without extension.
+    source = basename(filepath).split('.')[0];
+    const splitFilepath = filepath.split('/');
+    // Get the library name from the path.
+    const nodeModulesIndex = splitFilepath.indexOf('node_modules');
+    if (nodeModulesIndex !== -1) {
+      // Remove all non-alphanumeric characters.
+      const libraryName = splitFilepath[nodeModulesIndex + 1].replace(
+        /\W/g,
+        ''
+      );
```

Comment:
I think it would be good to implement the same data sanitization for the file name (`source`) as well

Reply:
That's actually a very good remark!